analyzer/src/main/kotlin/AnalyzerResultBuilder.kt

```diff
@@ -40,7 +40,13 @@ class AnalyzerResultBuilder(private val curationProvider: PackageCurationProvide
     private val issues = sortedMapOf<Identifier, List<OrtIssue>>()
     private val dependencyGraphs = sortedMapOf<String, DependencyGraph>()
 
-    fun build() = DependencyGraphConverter.convert(AnalyzerResult(projects, packages, issues, dependencyGraphs))
+    fun build(): AnalyzerResult {
+        require(projects.map { it.id }.toSet().intersect(packages.map { it.pkg.id }.toSet()).isEmpty()) {
```

Comment:
Could be `mapTo(mutableSetOf())` in both places.

Reply:
Maybe use `(projects.map { it.id } + packages.map { it.pkg.id }).getDuplicates()` to also get the actual set of packages that are also projects.